.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,56 @@
+{
+	"name": "Tensorflow Dev Container",
+	
+	"dockerFile": "Dockerfile",
+
+	// Uncomment this for GPU image
+	// "build": { 
+	// 		"args": { 
+	//			"IMAGE_TYPE": "-gpu"
+	//		} 
+	//	},
+	
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash",
+		"python.testing.pytestEnabled": true,
+		"python.testing.pytestArgs": [ 
+			"./tensorflow"
+		],
+		"C_Cpp.clang_format_style": "{BasedOnStyle: Google}",
+		"C_Cpp.default.includePath": [
+			"${workspaceFolder}/**",
+        	],
+		"files.watcherExclude": {
+			"**/bazel-*/**": true
+		},
+		"search.exclude": {
+			"**/bazel-*/**": true
+		}
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"ms-python.python",
+		"ms-vscode.cpptools",
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "uname -a",
+    "mounts": [
+        "source=tensorflow_cache,target=/home/vscode/.cache/bazel/"
+      ],
+
+	// Uncomment when using a ptrace-based debugger like C++, Go, and Rust
+	"runArgs": [ 
+		"--cap-add=SYS_PTRACE",
+		// Uncomment this to enable Nvidia support
+		//"--runtime=nvidia",
+		"--security-opt",
+		"seccomp=unconfined" ],
```

Comment:
From a security POV this is scary.

Where is this container used? If only locally / on dev VM / on ephemeral hosts, then it should be ok.

Reply:
Yes It is for contributors working locally (e.g. preparing a PR) or in the cloud with Github Codespace if it is supported.
It is [in the documentation](https://code.visualstudio.com/docs/remote/devcontainerjson-reference).

We could also distribute this section safe commented but we could have an impact on usability when working on C++.

See also https://forums.docker.com/t/security-implications-of-cap-add-sys-ptrace-security-opt-seccomp-unconfined-needed-for-rr/49816

I don't know if @Chuxel or @chrmarti know more about this context for the minimal settings.

Reply:
Codespaces are also housed on ephemeral VMs for this exact reason. The VM is the security boundary since things like docker-in-docker require full `--privileged` which needs to also work in that environment. These containers are also not intended for production use more generally (which is why they're called development containers).

Reply:
Thanks @Chuxel. 
I've commented  by default `seccomp=unconfined` as it seems not required for `Docker >= 1.13`. 
See the default profile on `1.13` tag:
https://github.com/moby/moby/blob/1.13.x/profiles/seccomp/default.json#L650-L666

Reply:
Do you know what is the current Docker version in Codespaces VMs?

Reply:
They're kept up to date so Moby Engine 20.10 unless there was a hiccup due to a Docker/Moby during testing in which case it would be the version right before it. Much newer than 1.13 in either event.